apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notices.js

```diff
@@ -0,0 +1,186 @@
+import { ExternalLink, Notice } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createInterpolateElement, render, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useCanvas } from './use-canvas';
+import { useGlobalStylesConfig } from './use-global-styles-config';
+import './notice.scss';
+
+const trackEvent = ( eventName, isSiteEditor = true ) =>
+	recordTracksEvent( eventName, {
+		context: isSiteEditor ? 'site-editor' : 'post-editor',
+		blog_id: wpcomGlobalStyles.wpcomBlogId,
+	} );
+
+function GlobalStylesWarningNotice() {
+	const { globalStylesInUse } = useGlobalStylesConfig();
+
+	useEffect( () => {
+		if ( globalStylesInUse ) {
+			trackEvent( 'calypso_global_styles_gating_notice_view_canvas_show' );
+		}
+	}, [ globalStylesInUse ] );
+
+	if ( ! globalStylesInUse ) {
+		return null;
+	}
+
+	return (
+		<Notice status="warning" isDismissible={ false } className="wpcom-global-styles-notice">
+			{ createInterpolateElement(
+				__(
+					'Your site includes customized styles that are only visible to visitors after <a>upgrading to the Premium plan or higher</a>.',
+					'full-site-editing'
+				),
+				{
+					a: (
+						<ExternalLink
+							href={ wpcomGlobalStyles.upgradeUrl }
+							target="_blank"
+							onClick={ () =>
+								trackEvent( 'calypso_global_styles_gating_notice_view_canvas_upgrade_click' )
+							}
+						/>
+					),
+				}
+			) }
+		</Notice>
+	);
+}
+
+function GlobalStylesViewNotice() {
+	const { canvas } = useCanvas();
+
+	useEffect( () => {
+		if ( canvas !== 'view' ) {
+			return;
+		}
+
+		const footer = document.querySelector( '.edit-site-sidebar__footer' );
+		if ( ! footer ) {
+			return;
+		}
+
+		const noticeContainer = document.createElement( 'div' );
+		footer.prepend( noticeContainer );
+		render( <GlobalStylesWarningNotice />, noticeContainer );
+	}, [ canvas ] );
+
+	return null;
+}
+
+function GlobalStylesEditNotice() {
+	const NOTICE_ID = 'wpcom-global-styles/gating-notice';
+	const { globalStylesInUse, globalStylesId } = useGlobalStylesConfig();
+	const { canvas } = useCanvas();
+	const { isSiteEditor, isPostEditor } = useSelect(
+		( select ) => ( {
+			isSiteEditor: !! select( 'core/edit-site' ) && canvas === 'edit',
+			isPostEditor: ! select( 'core/edit-site' ) && !! select( 'core/editor' ).getCurrentPostId(),
+		} ),
+		[ canvas ]
+	);
+
+	const { createWarningNotice, removeNotice } = useDispatch( 'core/notices' );
+	const { editEntityRecord } = useDispatch( 'core' );
+
+	const upgradePlan = useCallback( () => {
+		window.open( wpcomGlobalStyles.upgradeUrl, '_blank' ).focus();
+		trackEvent( 'calypso_global_styles_gating_notice_upgrade_click', isSiteEditor );
+	}, [ isSiteEditor ] );
+
+	const previewSite = useCallback( () => {
+		window.open( wpcomGlobalStyles.previewUrl, '_blank' ).focus();
+		trackEvent( 'calypso_global_styles_gating_notice_preview_click', isSiteEditor );
```

Comment:
Maybe we should add the post id as an extra prop? I **think** when you add a different extra prop, the event would be considered unique. This way, a user clicking from two posts will be considered two different clicks. 

The same thing should be for `calypso_global_styles_gating_notice_show` too; we want to register the event twice if the user saw the upgrade notice from two different posts/pages.

Reply:
I think "uniques" refers to distinct users, not events. See PCYsg-5hU-p2:

> Uniques – the number of distinct users associated with an event has been recorded in a timeframe

See also `/wp-content/lib/nosara/tracks-trends.php#281`